src/commands/deployments/disconnectRepo.ts

```diff
@@ -3,13 +3,20 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DeploymentsTreeItem } from "vscode-azureappservice";
+import { DeploymentsTreeItem, disconnectRepo as disconnectRepository } from "vscode-azureappservice";
 import { IActionContext } from "vscode-azureextensionui";
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
 import { ext } from "../../extensionVariables";
+import { localize } from '../../localize';
 
 export async function disconnectRepo(context: IActionContext, node?: DeploymentsTreeItem): Promise<void> {
     if (!node) {
         node = await ext.tree.showTreeItemPicker<DeploymentsTreeItem>(DeploymentsTreeItem.contextValueConnected, { ...context, suppressCreatePick: true });
     }
-    await node.disconnectRepo(context);
+
+    if (node.parent instanceof SiteTreeItem) {
```

Comment:
I do think this one might be worth keeping though since DeploymentsTreeItems will exist under the TrialApp, so it seems likely that we might accidentally pass up the wrong treeItem.